mathext/ell_carlson.go

```diff
@@ -0,0 +1,154 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext
+
+import (
+	"math"
+)
+
+// EllipticRF computes the symmetric elliptic integral R_F(x,y,z):
+//  R_F(x,y,z) = (1/2)\int_{0}^{\infty}{1/s(t)} dt,
+//  s(t) = \sqrt{(t+x)(t+y)(t+z)}.
+//
+// The arguments x, y, z must satisfy the following conditions, otherwise the function returns math.NaN():
+//  0 ≤ x,y,z ≤ upper,
+//  lower ≤ x+y,y+z,z+x,
+// where:
+//  lower = 5/(2^1022) = 1.112536929253601e-307,
+//  upper = (2^1022)/5 = 8.988465674311580e+306.
+//
+// The definition of the symmetric elliptic integral R_F can be found in NIST Digital Library of Mathematical Functions
+// (http://dlmf.nist.gov/19.16.E1). The original Fortran code was published as Algorithm 577 in ACM TOMS (http://doi.org/10.1145/355958.355970).
+// This code is also available as a part of SLATEC Common Mathematical Library (http://netlib.org/slatec/index.html). Later, Carlson described
+// an improved version in http://dx.doi.org/10.1007/BF02198293 (also available at https://arxiv.org/abs/math/9409227).
+func EllipticRF(x, y, z float64) float64 {
+	const (
+		lower = 5.0 / (1 << 256) / (1 << 256) / (1 << 256) / (1 << 254) // 5*2^-1022
+		upper = 1 / lower
+		tol   = 1.2674918778210762260320167734407048051023273568443e-02 // (3ε)^(1/8)
```

Comment:
I don't understand where this value of `tol` is coming from, it's not the same as in Carlson.